src/pages/ServiceDetails/ServiceInfo/types/Subset.ts

```diff
@@ -0,0 +1,31 @@
+import { TrafficPolicy } from '../../../../types/IstioObjects';
+import Label from './Label';
+
+export default class Subset {
+  name: string;
+  labels: { [key: string]: string };
+  trafficPolicy: TrafficPolicy;
+
+  constructor(name: string, labels: { [key: string]: string }, trafficPolicy: TrafficPolicy) {
+    this.name = name;
+    this.labels = labels;
+    this.trafficPolicy = trafficPolicy;
+  }
+
+  public isValid() {
+    return this.hasValidName() && this.hasValidLabels() && this.hasValidTrafficPolicy();
+  }
+
+  hasValidName() {
+    return typeof this.name === 'string';
+  }
+
+  hasValidLabels() {
+    const valid = Object.keys(this.labels).every((k, i) => new Label(k, this.labels[k]).isValid());
```

Comment:
I wonder if we can use an aux function instead to create a new "Label" class/abstraction just to perform a validation.
It sounds to me that we need to check a condition between key: k and value: this.labels[k].